modules/xfeatures2d/src/affine_feature2d.cpp

```diff
@@ -0,0 +1,681 @@
+            //Transformation
+            Mat warpedImgRoi;
+            warpAffine(img_roi, warpedImgRoi, transf, Size(int(maxx), int(maxy)),INTER_AREA, BORDER_REPLICATE);
+
+            //Point in U-Normalized coordinates
+            c = U * p;
+            cx = int(c(0, 0) - minx);
+            cy = int(c(1, 0) - miny);
+
+            if (warpedImgRoi.rows > 2 * radius+1 && warpedImgRoi.cols > 2 * radius+1)
+            {
+                //Cut around normalized patch
+                roix = std::max(cx - int(ceil(radius)), 0);
+                roiy = std::max(cy - int(ceil(radius)), 0);
+                roi = Rect(roix, roiy,
+                        cx - roix + std::min(int(ceil(radius)), warpedImgRoi.cols - cx-1)+1,
+                        cy - roiy + std::min(int(ceil(radius)), warpedImgRoi.rows - cy-1)+1);
+                warpedImg = warpedImgRoi(roi);
+
+                //Coordinates in cutted ROI
+                cx = cx - roix;
+                cy = cy - roiy;
+            } else
+                warpedImgRoi.copyTo(warpedImg);
+
+            //Integration Scale selection
+            si = selIntegrationScale(warpedImg, si, Point(cx, cy));
+            //Differentation scale selection
+            selDifferentiationScale(warpedImg, Lxm2smooth, Lxmysmooth, Lym2smooth, si,
+                    Point(cx, cy));
+
+            //Spatial Localization
+            cxPr = cx; //Previous iteration point in normalized window
+            cyPr = cy;
+
+            float cornMax = 0;
+            for (int j = 0; j < 3; j++)
+            {
+                for (int t = 0; t < 3; t++)
+                {
+                    float dx2 = Lxm2smooth.at<float> (cyPr - 1 + j, cxPr - 1 + t);
+                    float dy2 = Lym2smooth.at<float> (cyPr - 1 + j, cxPr - 1 + t);
+                    float dxy = Lxmysmooth.at<float> (cyPr - 1 + j, cxPr - 1 + t);
+                    float det = dx2 * dy2 - dxy * dxy;
+                    float tr = dx2 + dy2;
+                    float cornerness = det - (0.04f * tr*tr);
+                    if (cornerness > cornMax)
+                    {
+                        cornMax = cornerness;
+                        cx = cxPr - 1 + t;
+                        cy = cyPr - 1 + j;
+                    }
+                }
+            }
+
+            //Transform point in image coordinates
+            p(0, 0) = float(px);
+            p(1, 0) = float(py);
+            //Displacement vector
+            c(0, 0) = float(cx - cxPr);
+            c(1, 0) = float(cy - cyPr);
+            //New interest point location in image
+            p = p + Matx22f(Matx22d(U).inv()) * c;
+            px = int(p(0, 0));
+            py = int(p(1, 0));
+
+            q = calcSecondMomentSqrt(Lxm2smooth, Lxmysmooth, Lym2smooth, Point(cx, cy), Mk);
+
+            float ratio = 1 - q;
+
+            //if ratio == 1 means q == 0 and one axes equals to 0
+            if (!isnan(ratio) && ratio != 1)
```

Comment:
Should this be using `std::isnan()` similar to [other places](https://github.com/opencv/opencv_contrib/blob/246ea8f3bdf174a2aad6216c2601e3a93bf75c29/modules/sfm/src/libmv_light/libmv/numeric/numeric.h#L461) in the codebase? I ran into an error locally until I switched to that, though I'm still hitting other issues with the `master` branch, so I can't confirm that it solved everything.